remote_mapper.go

```diff
@@ -0,0 +1,196 @@
+	Interval        int64    `json:",omitempty"`
+	ChunkSize       int      `json:",omitempty"`
+}
+
+// Responses get streamed back to the remote mapper from the remote machine that runs a local mapper
+type MapResponse struct {
+	Err       string `json:",omitempty"`
+	Data      []byte
+	Completed bool `json:",omitempty"`
+}
+
+// Open is a no op, real work is done starting with Being
+func (m *RemoteMapper) Open() error { return nil }
+
+// Close the response body
+func (m *RemoteMapper) Close() {
+	if m.resp != nil && m.resp.Body != nil {
+		m.resp.Body.Close()
+	}
+}
+
+// Begin sends a request to the remote server to start streaming map results
+func (m *RemoteMapper) Begin(c *influxql.Call, startingTime int64, chunkSize int) error {
+	// get the function for unmarshaling results
+	f, err := influxql.InitializeUnmarshaller(c)
+	if err != nil {
+		return err
+	}
+	m.unmarshal = f
+
+	if c != nil {
+		m.Call = c.String()
+	}
+	m.ChunkSize = chunkSize
+	m.TMin = startingTime
+
+	// send the request to map to the remote server
+	b, err := json.Marshal(m)
+	if err != nil {
+		return err
+	}
+
+	// request to start streaming results
+	resp, err := http.Post(m.dataNodes[0].URL.String()+"/run_mapper", "application/json", bytes.NewReader(b))
+	if err != nil {
+		return err
+	}
+	m.resp = resp
+
+	return nil
+}
+
+// NextInterval is part of the mapper interface. In this case we read the next chunk from the remote mapper
+func (m *RemoteMapper) NextInterval() (interface{}, error) {
+	// just return nil if the mapper has completed its run
+	if m.complete {
+		return nil, nil
+	}
+
+	// read the chunk
+	chunk := make([]byte, MAX_MAP_RESPONSE_SIZE, MAX_MAP_RESPONSE_SIZE)
+	n, err := m.resp.Body.Read(chunk)
+	if err != nil {
+		return nil, err
+	}
+	if n == 0 {
+		return nil, nil
+	}
+
+	// marshal the response
+	mr := &MapResponse{}
+	err = json.Unmarshal(chunk[:n], mr)
+	if err != nil {
+		return nil, err
+	}
+	if mr.Err != "" {
+		return nil, errors.New(mr.Err)
+	}
+
+	// if it's a complete message, we've emptied this mapper of all data
+	if mr.Completed {
+		m.complete = true
+		return nil, nil
+	}
+
+	// marshal the data that came from the MapFN
+	v, err := m.unmarshal(mr.Data)
+	if err != nil {
+		return nil, err
+	}
+
+	return v, nil
+}
+
+// CallExpr will parse the Call string into an expression or return nil
+func (m *RemoteMapper) CallExpr() (*influxql.Call, error) {
+	if m.Call == "" {
+		return nil, nil
+	}
+
+	c, err := influxql.ParseExpr(m.Call)
+	if err != nil {
+		return nil, err
+	}
+	call, ok := c.(*influxql.Call)
+
+	if !ok {
+		return nil, errors.New("Could't marshal aggregate call")
+	}
+	return call, nil
+}
+
+// FilterExprs will parse the filter strings and return any expressions. This array
+// will be the same size as the SeriesIDs array with each element having a filter (which could be nil)
+func (m *RemoteMapper) FilterExprs() []influxql.Expr {
+	exprs := make([]influxql.Expr, len(m.SeriesIDs), len(m.SeriesIDs))
```

Comment:
I think he's got it right - he's just preallocating the slice and walking it in the loops below. However, I think it can just be shortened to:
```
exprs := make([]influxql.Expr, len(m.SeriesIDs))
```